mkosi/resources/mkosi-initrd/mkosi.profiles/network/mkosi.conf

```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+[Match]
+Distribution=!arch
+
+[Content]
+Packages=systemd-resolved
```

Comment:
Let's do this a bit more structured, I'd leave mkosi.conf empty and just have mkosi.conf.d/ with two files: systemd-networkd.conf and systemd-resolved.conf, each with the appropriate matches in them.